pkg/workspace/plugins.go

```diff
@@ -631,12 +631,17 @@ func getCandidateExtensions() []string {
 	return []string{""}
 }
 
-// pluginRegexp matches plugin filenames: pulumi-KIND-NAME-VERSION[.exe].
+// pluginRegexp matches plugin directory names: pulumi-KIND-NAME-VERSION.
 var pluginRegexp = regexp.MustCompile(
 	"^(?P<Kind>[a-z]+)-" + // KIND
 		"(?P<Name>[a-zA-Z0-9-]*[a-zA-Z0-9])-" + // NAME
 		"v(?P<Version>.*)$") // VERSION
 
+// installingPluginRegexp matches the name of folders for plugins which are being installed. During installation
+// we extract plugins to a folder with a suffix of `.tmpXXXXXX` (where `XXXXXX`) is a random number, from
+// ioutil.TempFile. We should ignore these plugins as they have not yet been successfully installed.
+var installingPluginRegexp = regexp.MustCompile(`\.tmp[0-9]+$`)
```

Comment:
Is the `[0-9]+` portion guaranteed to be present, or might it be absent on a machine which has never installed plugins?

Reply:
> Is the [0-9]+ portion guaranteed to be present, or might it be absent on a machine which has never installed plugins?

`ioutil.TempFile` is documented as adding a random suffix, and the implementation always adds a random integer to the end (even if just `.tmp` would be unique).

The fact that suffix is all numbers is not explicitly documented by `ioutil.TempFile`, but that's the implementation that the go stdlib has today.